drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -117,6 +117,42 @@ static struct stm32_i2c_timings_t i2c_valid_timing[STM32_I2C_VALID_TIMING_NBR];
 static uint32_t i2c_valid_timing_nbr;
 #endif /* CONFIG_I2C_STM32_V2_TIMING */
 
+#ifdef CONFIG_I2C_DMA
+static int configure_dma(const struct device *dma_dev, uint32_t dma_channel,
+			 struct dma_config *dma_cfg, struct dma_block_config *blk_cfg,
+			 uint32_t source_address, uint32_t dest_address, struct i2c_msg *msg,
+			 struct k_sem *dma_sem, const char *direction)
```

Comment:
It is suboptimal to use more than 4 parameters in a function on ARM platforms. 4 first params are accessed from registers (r0-r3), other params will be accessed from stack.
Any way to do this differently ?

Reply:
structs re-jigged